api/src/main.rs

```diff
@@ -17,25 +17,15 @@ async fn main() {
     env_logger::init();
     log::info!("Starting service");
 
-    let configuration = {
-        dotenv().ok();
-        match Configuration::init_from_env() {
-            Ok(val) => val,
-            Err(e) => {
-                log::error!("{}", e);
-                log::error!("Terminating because of previous error.");
-                process::exit(1);
-            }
-        }
-    };
-    match CONFIGURATION.set(configuration) {
-        Ok(()) => log::info!("Successfully provided global service configuration"),
-        Err(_) => {
-            log::error!("Failed to provide global service configuration");
-            log::error!("Terminating because of previous error.");
-            process::exit(1);
-        }
-    }
+    dotenv().ok();
+    let conf = Configuration::init_from_env()
+        .map_err(log_error_and_panic)
+        .unwrap();
+
+    CONFIGURATION
+        .set(conf)
+        .map_err(|_| log_error_and_panic("Failed to provide global service configuration"))
```

Comment:
In that case we are not using the value of the error, which I think we should.

Probably it would be good to have 2 methods
```rust
fn log_unwrap(e: impl std::fmt::Display) {
    // ...
}

fn log_expect(e: impl std::fmt::Display, msg: &str) {
    // ...
}
```

Reply:
The problem with `once_cell` and the set method is, that it does not return an actual error. The content of the error is the configuration struct. This is why I did this way. I'm not really happy with this solution, but as we panic the program anyway it is also not a problem that we do not use it.

Reply:
We can use two function for sure. Do we want to have logs  for successful unwrap?